action.yaml

```diff
@@ -264,7 +264,7 @@ runs:
     - name: Get Deployment Type
       run: |
         cd ${{ inputs.root-folder }}
-        files=$(git diff --name-only HEAD^..HEAD)
+        files=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }})
```

Comment:
I am not sure if this will work with this change alone, because the clone on line 90 only fetches with a depth of two commits. Some options may be to fetch a reasonably large number of commits, say 50, which leaves a small risk of missing some in extreme cases, or to fetch all commits (I think the default if you don't specify fetch-depth, would need to be check that) which could be slow on very large repos, or to somehow use the push event payload to figure out the required fetch depth for that push (not sure if this is possible, but it would be the length of the commits list + 1).

Reply:
> Only a single commit is fetched by default, for the ref/SHA that triggered the workflow. Set fetch-depth: 0 to fetch all history for all branches and tags. Refer [here](https://docs.github.com/actions/using-workflows/events-that-trigger-workflows) to learn which commit $GITHUB_SHA points to for different events.

https://github.com/actions/checkout

Reply:
Could we just run `git fetch origin ${{ github.event.before }}` after checkout?